### DISCUSSION TITLE
Is keepUnusedDataFor property won't work for the already mounted component?

### DISCUSSION THREAD

**Discussion by mavericksab81** [reduxjs/redux-toolkit#3041] [General]
I have a get call which I am dispatching on the component when mounted. But, there is requirement where I have to refetch after some time elapsed, ex- 30 seconds.
But, as per the docs, keepUnusedDataFor works only when the component is unmounted and mounted again within the time frame.

So, currently, I am doing conditional refetech after the last fulfilledTimeStamp and currentTimestamp diff. Is there any better way to do that.
If we do using the refetchOnMountOrArgChange, then also the situation remains the same.

Thoughts pls.

**Comment by phryneas** [comment]
I think you want to use [polling](https://redux-toolkit.js.org/rtk-query/usage/polling)

**Reply by mavericksab81** [reply]
Yes, I was looking something for this option only.
